.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+              } else {
+                console.log('No online runner with label "ARM64" found, using default runner');
+              }
+            } else {
+              console.log('Platform is not linux/arm64, using default X64 runner');
+            }
+            core.setOutput('runner_label', runner_label);
+
+      - name: Overwrite runner label
+        id: set_runner_label
+        run: |
+          if [ -n "${{ steps.determine_runner.outputs.runner_label }}" ]; then
+            echo "runner_label=${{ steps.determine_runner.outputs.runner_label }}" >> $GITHUB_OUTPUT
+          else
+            echo "runner_label=${{ steps.set_default_runner_label.outputs.runner_label }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Debug variables before build
+        if: steps.set_proceed_flag.outputs.proceed_valid == 'true'
+        run: |
+          echo "Context: ${{ github.workspace }}/${{ steps.parse_path.outputs.directory }}"
+          echo "File: ${{ github.workspace }}/${{ steps.changed_files.outputs.files }}"
+          echo "Tag: ${{ vars.DOCKERHUB_USERNAME }}/${{ steps.parse_path.outputs.dockerfile_name }}-${{ steps.parse_path.outputs.platform_tag }}:${{ steps.date.outputs.date }}"
+          echo "Platforms: ${{ steps.parse_path.outputs.platform }}"
+          echo "Runner label: ${{ steps.determine_runner.outputs.runner_label }}"
+      
+
+      
+  BUILD-job:
+    needs: PREPARE-job
+    runs-on: ${{ needs.PREPARE-job.outputs.runner_label }}
+    if: needs.PREPARE-job.outputs.proceed_valid == 'true'
+    steps:
+      - name: Print hostname
+        run: |
+          echo "Hostname: $(hostname)"
+      
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1  # As the runs-on machine maybe different from Build, re-checkout source code. Only the current commit is needed
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: ${{ needs.PREPARE-job.outputs.platform }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker-container
+          install: true
+    
+      - name: Show current Buildx builders
+        run: docker buildx ls
+
+      - name: Enable BuildKit
+        run: |
+          echo ${{ needs.PREPARE-job.outputs.runner_label}} 
+          echo "DOCKER_BUILDKIT=1" >> $GITHUB_ENV 
+
+      - name: Check and set docker cache location
+        id: docker_cache_check
+        run: |
+          CACHE_DIR="$HOME/runner/docker-cache"
+          echo "CACHE_DIR=$CACHE_DIR"
+
+          if [ -d "$CACHE_DIR" ]; then
+            echo "Cache directory exists."
+            CACHE_SIZE=$(du -sh "$CACHE_DIR" | cut -f1)
+            echo "Cache directory size: $CACHE_SIZE"
+          else
+            echo "Cache directory does not exist. Creating..."
+            sudo mkdir -p "$CACHE_DIR"
```

Comment:
Do we need any logic in case this fails? I don't think so but just raising the question.

Reply:
Repaired in the submission. Thanks.